examples/save_disp.py

```diff
@@ -1,67 +1,103 @@
+
+    return parser.parse_args(arg_list)
+
+
+def main() -> None:
+    args = parse_args()
+
+    assert os.path.exists(args.model), f"Given model file does not exist: {args.model}"
+    assert os.path.exists(args.left_image), (f"Left image path does not exist: {args.left_image}")
+    assert os.path.exists(args.right_image), (f"Right image path does not exist: {args.right_image}")
+
     # Turn off deprecated nvfuser if Torch version is old.
-    # Displays multiple warnings otherwise.
+    # Displays warning messages otherwise.
     if version.parse(torch.__version__) < version.parse("2.2.0"):
         torch._C._jit_set_nvfuser_enabled(False)
 
     with torch.no_grad():
-        model_path = Path("output") / "monster-mix-script.pt"
-        assert os.path.exists(model_path), f"Model file not found at {model_path}"
+        device = "cuda" if torch.cuda.is_available() else "cpu"
 
-        model = torch.jit.load(model_path, map_location="cuda")
+        model = torch.jit.load(args.model, map_location=device)
         model.eval()
 
         print("TorchScript model loaded")
 
-        image_left_path = Path("input") / "left" / "sample_l.png"
-        image_right_path = Path("input") / "right" / "sample_r.png"
-
-        assert os.path.exists(image_left_path), (
-            f"Left image not found at {image_left_path}"
-        )
-        assert os.path.exists(image_right_path), (
-            f"Right image not found at {image_right_path}"
-        )
-
-        image_left = load_image(image_left_path)
-        image_right = load_image(image_right_path)
+        image_left = load_image(args.left_image, device)
+        image_right = load_image(args.right_image, device)
 
         print("Stereo image pair loaded")
 
-        disparity = model(image_left, image_right)
-        torch.cuda.synchronize()
+        # Note: Use two warm up runs in production.
 
-        disparity = disparity.cpu()
-        disparity = disparity.numpy()
+        # Predict the disparity for a given stereo image pair.
+        disparity = model(image_left, image_right)
+        disparity = disparity.cpu().numpy()
 
         # Scaling by 256 done to minimize precision loss when saving an image.
         # TODO: Normalize values in the range [0, 2^16)
         disparity_image = np.round(disparity * 256).astype(np.uint16)
-        # Save the disparity image.
-        output_path = Path("output") / "disparity.png"
-        Image.fromarray(disparity_image).save(output_path)
 
-        print(f"Image saved to {output_path}")
+        # Create missing directories to save the disparity image.
+        split_out_path = args.disparity_path.parts
+        if len(split_out_path) > 1:
+            os.makedirs(os.path.join(*split_out_path[:-1]), exist_ok=True)
+
+        # Save the disparity image.
+        Image.fromarray(disparity_image).save(args.disparity_path)
+        print(f"Image saved to {args.disparity_path}")
```

Comment:
Use `pathlib` to your advantage.
```suggestion
        # Save the disparity image.
        args.disparity_path.parent.mkdir(parents=True, exist_ok=True)
        Image.fromarray(disparity_image).save(args.disparity_path)
        print(f"Image saved to {args.disparity_path}")
```